types/dec_coin.go

```diff
@@ -169,6 +180,11 @@ func NewDecCoins(coins Coins) DecCoins {
 	return decCoins
 }
 
+// AddDecCoin adds a new decimal coin to the existed set of decimal coins.
+func (coins DecCoins) AddDecCoin(decCoin DecCoin) DecCoins {
```

Comment:
This function needs to have the same checks as `NewCoins`. Technically it's duplicate of `func (coins DecCoins) Add(coinsB DecCoins) DecCoins` (L227) with the difference that this only appends a single coin without validation.

Ideally, it could be changed to `Add(coinsB ...DecCoin) DecCoins` to support both cases.

Reply:
I'm good with the rewrite of the `Add` method and the remove of the `AddDecCoin`, but it can be a change on the library usage so it can break third-party systems who using this. I'm not familiar enough with this to make this decision.

Reply:
> I'm good with the rewrite of the Add method and the remove of the AddDecCoin, but it can be a change on the library usage so it can break third-party systems who using this.

It's fine as long as it's documented as an API breaking change on the CHANGELOG